### DISCUSSION TITLE
Unique slugs

### DISCUSSION THREAD

**Discussion by tsekka** [area17/twill#1381] [Q&A] [answered]
Hi there,

Is there a quick, straightforward way to enable only unique slugs? 

Thanks.

**Comment by haringsrob** [comment]
Hey @tsekka,

Out of the box there is non as far as I can see.

@ifox was there ever something added for this?

To integrate this there will need to be some changes. And this may be more difficult than expected because the full slug is not stored. Only that after the permalink

- [ ] On the front-end we have to do a check when entering a sluggable model title instead of just adding it
- [ ] Actually check based on the permalink if there is already a matching slug. Maybe scope it to the current model?

**Comment by ifox** [comment]
Hi guys,

I'm not sure I'm following. Is this about slug uniqueness across multiple modules? Within one module, slugs are already uniques (in combination with their locale).

**Comment by haringsrob** [comment]
I do not think they are unique.

<img width="423" alt="Screenshot 2022-01-19 at 14 38 15" src="https://user-images.githubusercontent.com/866743/150141645-48bf4d98-181c-471a-a45f-6cdd4efe29c2.png">

**Reply by ifox** [reply]
This looks like a regression, see https://github.com/area17/twill/blob/2.x/src/Models/Behaviors/HasSlug.php#L124

**Reply by haringsrob** [reply]
I will write a test and look to fix this :)

**Reply by ifox** [reply]
Ok this is a 2.6.0 regression due to https://github.com/area17/twill/commit/99ffa44dc99c04349420889758eee6604868f774#diff-522aa2fdc5f1c037470707e1aa775f40445c9bc9dbf802f02e267975a63281d6.

**Comment by haringsrob** [comment, marked as answer]
Fix and test added: https://github.com/area17/twill/pull/1390
